### ISSUE TITLE
Lint for Unnecessary Map

### ISSUE THREAD

**Issue by mmstick** [rust-lang/rust-clippy#1005]
It woud be nice if clippy could detect situations where `map()` is being used unnecessarily to pass a contained value wrapped as either a Result or an Option.
### Unnecessarry

``` rust
let mut file = File::open(path)
         .map(|file| file)
         .unwrap_or_else(|why| panic!("couldn't open {}: {}", infilename, why));
```
### Recommended

``` rust
let mut file = File::open(infilename)
         .unwrap_or_else(|why| panic!("couldn't open {}: {}", infilename, why));
```

**Comment by shepmaster** [comment]
> wrapped as either a Result or an Option.

Wouldn't this apply for wider uses of `map`, such as on an iterator?

**Comment by mmstick** [comment]
> Wouldn't this apply for wider uses of `map`, such as on an iterator?

I suppose it would also apply for using `map(|x| x)` uselessly on an `Iterator` too.

**Comment by camsteffen** [comment]
Resolved by #5694
